src/customerio-inapp.ts

```diff
@@ -45,18 +50,40 @@ class CustomerIOInAppMessaging implements NativeSpec {
       listener(event);
     };
 
-    return withNativeModule(async (native) => {
-      // Old arch requires async return via Promise; new arch supports sync
-      const isNewArch = await native.isNewArchEnabled();
-      if (isNewArch) {
-        // If the new architecture is enabled, use the TurboModule's event emitter
-        return native.onInAppEventReceived(emitter);
+    // Holds the real subscription once created
+    let actualSubscription: EventSubscription | EmitterSubscription;
+    let removed = false;
+
+    // Proxy object returned immediately to support .remove()
+    const proxySubscription = {
+      remove: () => {
+        removed = true;
+        actualSubscription?.remove();
+      },
+    };
```

Comment:
The workaround avoids breaking changes for existing customers and works fine across both architectures